conjur/wrapper/keystore_adapter.py

```diff
@@ -31,52 +32,70 @@ def set_password(cls, identifier, key, val):
+                                      f"{identifier}") from keyring_error
 
     @classmethod
     def get_password(cls, identifier, key):
         """
         Method for getting a password in keyring
         """
-        return keyring.get_password(identifier, key)
+        try:
+            return keyring.get_password(identifier, key)
+        except keyring.errors.KeyringError as keyring_error:
+            raise KeyringAdapterGeneralError(f"unable to get value for key: {key} "
+                                      f"for identifier: {identifier}") from keyring_error
 
-    # TODO add simple delete for one attribute only here in the adapter
     # pylint: disable=try-except-raise
     @classmethod
-    def delete_password(cls, identifier):
+    def delete_password(cls, identifier, key):
         """
         Method for deleting a password in keyring
         """
         try:
-            for attr in KEYSTORE_ATTRIBUTES:
-                keyring.delete_password(identifier, attr)
-        # TODO do not raise keyring module errors
-        # Catches when credentials do not exist in the keyring. If the key does not exist,
-        # the user has already logged out
-        except keyring.errors.PasswordDeleteError:
-            return
-        except keyring.errors.KeyringError:
-            raise
+            keyring.delete_password(identifier, key)
+        except keyring.errors.PasswordDeleteError as password_error:
+            raise KeyringAdapterDeletionError(f"unable to delete key: {key} for identifier: "
+                                       f"{identifier}") from password_error
+        except keyring.errors.KeyringError as keyring_error:
+            raise KeyringAdapterGeneralError(f"unable to delete key: {key} for identifier:"
+                                      f" {identifier}") from keyring_error
 
     @classmethod
     def get_keyring_name(cls):
         """
         Method to get the system's keyring name
         """
-        return keyring.get_keyring().name
+        # keyring.get_keyring() can throw various types of exceptions
+        # some are OS exceptions that is the reason we catch general exception.
+        # please note that this is a critical path, if we get error here the
+        # entire app could not function
+        try:
+            return keyring.get_keyring().name
+        except Exception as err:  # pylint: disable=broad-except
+            logging.debug(err)
+            return None
 
-    # TODO add a more robust check
     @classmethod
     def is_keyring_accessible(cls):
         """
         Method to check if the keyring is accessible
         """
-        # TODO investigate other errors that would make a keyring not accessible
         try:
             # Send a dummy request to test if the keyring is accessible
+            # this should return None if value not exist
+            # catch Exception and not specific type since it can throw various
```

Comment:
this comment should be above the exceptions

"
Catch Exception and not a specific type since various exceptions can be thrown. For example some related to OS and others related to keyring.
"

Reply:
@mbenita-Cyberark